### PR TITLE
Pytest for merge loom test code

### PR DESCRIPTION
**Pull request by kishorikonwar**

### CODE REVIEW

**Review by khajoue2 — APPROVED** [review]

File: dockers/skylab/HCA_post_processing/Dockerfile [hunk 1]
```diff
@@ -8,6 +8,7 @@ COPY requirements.txt .
 RUN pip3 install numpy==1.17.0
 RUN pip3 install cython==0.29.15
 RUN pip3 install -r requirements.txt
+RUN pip3 install -U  pytest
```
Comment (jessicaway) [inline dockers/skylab/HCA_post_processing/Dockerfile:11]: I don't think we need to add the test (and especially the tests data) to the dockerfile. We can think about how to run these in an automated way at a later time, but I imagine we will do something like the sctools tests (which set up an environment and then run pytest). A this is here we would only test the code that is built into the docker, which is not necessarily useful for PR tests

File: dockers/skylab/HCA_post_processing/optimus_HCA_loom_merge.py [hunk 2]
```diff
@@ -1,9 +1,8 @@
-#!/usr/bin/env python3
-
 import argparse
 import loompy
 import numpy as np
-
+import tempfile
```
Comment (jessicaway) [inline dockers/skylab/HCA_post_processing/optimus_HCA_loom_merge.py:4]: No longer using these imports in this file

File: dockers/skylab/HCA_post_processing/optimus_HCA_loom_merge.py [hunk 3]
```diff
@@ -85,11 +87,12 @@ def main():
                 # filter out cells with low counts n_molecules > 1
                 UMIs = ds.ca['n_molecules']
                 cells = np.where(UMIs >= 100)[0]
-                for (ix, selection, view) in ds.scan(items=cells, axis=1, key="gene_names"):
+                for (ix, selection, view) in ds.scan(items=cells, axis=1): 
+                    view.ca['cell_names'] = view.ca['cell_names'] + "-" + str(i)
```
Comment (jessicaway) [inline dockers/skylab/HCA_post_processing/optimus_HCA_loom_merge.py:91]: did you remove the call that does this in the original file?

File: dockers/skylab/HCA_post_processing/test_optimus_HCA_loom_merge.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    import os
+    print('Test to merge three loom files')
+    # original file 
+    looma = os.path.join(_test_data_dir, "a.loom")
+    loomb = os.path.join(_test_data_dir, "b.loom")
+    loomc = os.path.join(_test_data_dir, "c.loom")
+    
+    # make a copy of these in the tmp file to avoid being overwritten
+    # during reapeated pytest runs on pytest
+    tmplooma = tmpdir.join("a.loom")
```
Comment (jessicaway) [inline dockers/skylab/HCA_post_processing/test_optimus_HCA_loom_merge.py:20]: No need to copy the files if we no longer change the original loom file in-place

File: pipelines/skylab/optimus/Optimus.changelog.md [hunk 5]
```diff
@@ -1,3 +1,10 @@
+# 4.1.9
+
+2020-12-7 (Date of Last Commit)
```
Comment (jessicaway) [inline pipelines/skylab/optimus/Optimus.changelog.md:3]: I don't think this should include any changes to optimus or SS2. The post processing pipeline is not a versioned pipeline

File: tasks/skylab/LoomUtils.wdl [hunk 6]
```diff
@@ -3,7 +3,7 @@ version 1.0
 task SmartSeq2LoomOutput {
   input {
     #runtime values
-    String docker = "quay.io/humancellatlas/secondary-analysis-loom-output:0.0.4-ss2-loom-fix-1"
+    String docker = "quay.io/humancellatlas/secondary-analysis-loom-output:0.0.5-fk"
```
Comment (jessicaway) [inline tasks/skylab/LoomUtils.wdl:6]: why did this change? None of the changes above are included in this docker image

**Review by jessicaway — CHANGES_REQUESTED** [review]

File: dockers/skylab/HCA_post_processing/Dockerfile (hunk 1 above)
Comment (kishorikonwar) [reply dockers/skylab/HCA_post_processing/Dockerfile:11]: removed.

File: dockers/skylab/HCA_post_processing/optimus_HCA_loom_merge.py (hunk 3 above)
Comment (kishorikonwar) [reply dockers/skylab/HCA_post_processing/optimus_HCA_loom_merge.py:91]: discussed and already removed.

File: dockers/skylab/HCA_post_processing/test_optimus_HCA_loom_merge.py (hunk 4 above)
Comment (kishorikonwar) [reply dockers/skylab/HCA_post_processing/test_optimus_HCA_loom_merge.py]: removed the steps to create temporary files

File: pipelines/skylab/optimus/Optimus.changelog.md (hunk 5 above)
Comment (kishorikonwar) [reply pipelines/skylab/optimus/Optimus.changelog.md]: agree and removed

File: tasks/skylab/LoomUtils.wdl (hunk 6 above)
Comment (kishorikonwar) [reply tasks/skylab/LoomUtils.wdl]: discussed

File: pipelines/skylab/smartseq2_multisample/MultiSampleSmartSeq2.wdl [hunk 7]
```diff
@@ -35,6 +35,7 @@ workflow MultiSampleSmartSeq2 {
       Boolean paired_end
   }
   # Version of this pipeline
+
```
Comment (jessicaway) [inline pipelines/skylab/smartseq2_multisample/MultiSampleSmartSeq2.wdl:38]: remove blank line

File: pipelines/skylab/smartseq2_single_sample/SmartSeq2SingleSample.wdl [hunk 8]
```diff
@@ -36,6 +36,7 @@ workflow SmartSeq2SingleCell {
   }
   
   # version of this pipeline
+
```
Comment (jessicaway) [inline pipelines/skylab/smartseq2_single_sample/SmartSeq2SingleSample.wdl:39]: same here

File: tasks/skylab/LoomUtils.wdl [hunk 9]
```diff
@@ -3,7 +3,9 @@ version 1.0
 task SmartSeq2LoomOutput {
   input {
     #runtime values
+
```
Comment (jessicaway) [inline tasks/skylab/LoomUtils.wdl:6]: also these

File: tasks/skylab/LoomUtils.wdl [hunk 10]
```diff
@@ -3,7 +3,9 @@ version 1.0
...
     String docker = "quay.io/humancellatlas/secondary-analysis-loom-output:0.0.5-fk-1"
+
```
Comment (jessicaway) [inline tasks/skylab/LoomUtils.wdl:7]:
```suggestion
```

File: tasks/skylab/LoomUtils.wdl [hunk 11]
```diff
@@ -61,7 +63,9 @@ task OptimusLoomGeneration {
 
   input {
     #runtime values
+
```
Comment (jessicaway) [inline tasks/skylab/LoomUtils.wdl:64]:
```suggestion
```

File: tasks/skylab/LoomUtils.wdl [hunk 12]
```diff
@@ -61,7 +63,9 @@ task OptimusLoomGeneration {
...
     String docker = "quay.io/humancellatlas/secondary-analysis-loom-output:0.0.5-fk-1"
+
```
Comment (jessicaway) [inline tasks/skylab/LoomUtils.wdl:66]:
```suggestion
```

File: tasks/skylab/LoomUtils.wdl [hunk 13]
```diff
@@ -149,7 +153,9 @@ task AggregateSmartSeq2Loom {
         String batch_id
         String? batch_name
         String pipeline_version
+
```
Comment (jessicaway) [inline tasks/skylab/LoomUtils.wdl:154]:
```suggestion
```

File: tasks/skylab/LoomUtils.wdl [hunk 14]
```diff
@@ -149,7 +153,9 @@ task AggregateSmartSeq2Loom {
...
         String docker = "quay.io/humancellatlas/secondary-analysis-loom-output:0.0.5-fk-1"
+
```
Comment (jessicaway) [inline tasks/skylab/LoomUtils.wdl:156]:
```suggestion
```

**Review by jessicaway — APPROVED** [review]

**Pull request merged**